static_frame/core/frame.py

```diff
@@ -5968,6 +6013,27 @@ def to_latex(self,
                 )
         return repr(self.display(config))
 
+    def to_msgpack(self) -> 'msgpack':
+        '''
+        Return a msgpack.
+        '''
+        import msgpack
+        import msgpack_numpy
+        def cast_msgpack(input):
+            try:
+                data = msgpack.packb(input, default=msgpack_numpy.encode) #try coercing standard and numpy datatypes
+            except TypeError:
+                #This inserts a magic character with any arbitrary datatype that msgpack_numpy doesn't support
+                #TODO: Is this attempting to handle too much? Is there a better way?
+                clsname = '>'+input[0].__class__.__module__+'.'+input[0].__class__.__name__                
+                data = msgpack.packb([clsname]+[a.__str__() for a in input], use_bin_type=True) #else cast to string
+            return data
+        return cast_msgpack({
+            '_index' : cast_msgpack([index for index in self._index]),
+            '_columns' : cast_msgpack([column for column in self._columns]),
+            '_name' : cast_msgpack(self._name),
+            '_blocks' : [cast_msgpack([value[0] for value in block.values]) for block in self.transpose()._blocks],
```

Comment:
Encoding at the block level is the right approach here, but why are we transposing first? Transposing may introduce undesirable type coercion. I would just try to encode each np.array in `self._blocks._blocks` (that is the underlying list of np.arrays).

Reply:
I used transpose out of my ignorance of static_frame's structure, self._blocks._blocks sounds like a much better approach

Reply:
updated functions to use _blocks._blocks. Undesirable coercion was occurring in test_frame_from_msgpack_c, everything was turned to float64 from the transposition.